packages/core/src/types/module-builder.ts

```diff
@@ -128,44 +128,149 @@ export interface SendDataOptions {
 
+  /**
+   * Deploys a contract.
+   *
+   * @param contractName - The name of the contract to deploy
+   * @param args - The arguments to pass to the contract constructor
+   * @param options - The options for the deployment
+   *
+   * @example
+   * ```
+   * const myContract = m.contract("MyContract", [], { value: 100 });
+   * ```
+   */
   contract<ContractNameT extends string>(
     contractName: ContractNameT,
     args?: ArgumentType[],
     options?: ContractOptions
   ): NamedArtifactContractDeploymentFuture<ContractNameT>;
 
+  /**
+   * Deploys a contract.
+   *
+   * @param contractName - The name of the contract to deploy
+   * @param artifact - The artifact of the contract to deploy
+   * @param args - The arguments to pass to the contract constructor
+   * @param options - The options for the deployment
+   *
+   * @example
+   * ```
+   * const myContract = m.contract("MyContract", [], { value: 100 });
+   * ```
+   */
   contract<const AbiT extends Abi>(
     contractName: string,
     artifact: Artifact<AbiT>,
     args?: ArgumentType[],
     options?: ContractOptions
   ): ContractDeploymentFuture<AbiT>;
 
+  /**
+   * Deploys a library.
+   *
+   * @param libraryName - The name of the library to deploy
+   * @param options - The options for the deployment
+   *
+   * @example
+   * ```
+   * const owner = m.getAccount(1);
+   * const myLibrary = m.library("MyLibrary", { from: owner } );
+   * ```
+   */
   library<LibraryNameT extends string>(
     libraryName: LibraryNameT,
     options?: LibraryOptions
   ): NamedArtifactLibraryDeploymentFuture<LibraryNameT>;
 
+  /**
+   * Deploys a library.
+   *
+   * @param libraryName - The name of the library to deploy
+   * @param artifact - The artifact of the library to deploy
+   * @param options - The options for the deployment
+   *
+   * @example
+   * ```
+   * const owner = m.getAccount(1);
+   * const myLibrary = m.library("MyLibrary", myLibraryArtifact, { from: owner });
+   * ```
+   */
   library<const AbiT extends Abi>(
     libraryName: string,
     artifact: Artifact<AbiT>,
     options?: LibraryOptions
   ): LibraryDeploymentFuture<AbiT>;
 
+  /**
+   * Calls a contract function.
+   *
+   * @param contractFuture - The contract to call
+   * @param functionName - The name of the function to call
+   * @param args - The arguments to pass to the function
+   * @param options - The options for the call
+   *
+   * @example
+   * ```
+   * const myContract = m.contract("MyContract");
+   * const myContractCall = m.call(myContract, "updateCounter", [100]);
+   * ```
+   */
   call<ContractNameT extends string, FunctionNameT extends string>(
     contractFuture: CallableContractFuture<ContractNameT>,
     functionName: FunctionNameT,
     args?: ArgumentType[],
     options?: CallOptions
   ): ContractCallFuture<ContractNameT, FunctionNameT>;
 
+  /**
+   * Statically calls a readonly contract function and returns the result.
```

Comment:
We should add a bit of explanation on the the following line about what a static call is, and that it won't update the blockchain.